src/odata-to-abstract-sql.ts

```diff
@@ -735,38 +735,31 @@ export class OData2AbstractSQL {
 		}
 	}
 	AddSelectFields(path: any, query: Query, resource: Resource) {
-		let fields;
+		let odataFieldNames: Array<[Resource, string]>;
 		if (
 			path.options &&
 			path.options.$select &&
 			path.options.$select.properties
 		) {
 			this.AddExtraFroms(query, resource, path.options.$select.properties);
-			fields = path.options.$select.properties.map((prop: any) =>
-				this.Property(prop),
-			);
-			fields = _(fields)
-				.reject(field =>
-					_.some(
-						query.select,
-						existingField => _.last(existingField) === field.name,
-					),
-				)
-				.map(field => this.AliasSelectField(field.resource, field.name))
-				.value();
+			odataFieldNames = path.options.$select.properties.map((prop: any) => {
+				const field = this.Property(prop) as {
+					resource: Resource;
+					name: string;
+				};
+				return [field.resource, field.name];
+			});
 		} else {
-			const resourceMapping = this.ResourceMapping(resource);
-			fields = _(resourceMapping)
-				.keys()
-				.reject(fieldName =>
-					_.some(
-						query.select,
-						existingField => _.last(existingField) === fieldName,
-					),
-				)
-				.map(field => this.AliasSelectField(resource, field))
-				.value();
+			odataFieldNames = resource.fields.map(({ fieldName }) => [
+				resource,
+				sqlNameToODataName(fieldName),
+			]);
 		}
+		const fields = _.differenceWith(
+			odataFieldNames,
+			query.select,
+			(a, b) => a[1] === _.last(b),
```

Comment:
This matches the previous
```ts
				.reject(field =>
					_.some(
						query.select,
						existingField => _.last(existingField) === field.name,
					),
				)
```
logic that was in both branches and could in fact use that exact code, but `_.differenceWith` is cleaner imo